ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,346 @@
+
+                        if (finalFileExists)
+                        {
+                            Log.Error($"BenzingaDataConverter.WriteToFile(): Moving backup file: {existingReferenceFileBackupPath} back to original location: {referenceFilePath}");
+                            File.Move(existingReferenceFileBackupPath, referenceFilePath);
+                        }
+
+                        continue;
+                    }
+
+                    // Clean up after ourselves so that we don't accidentally run out of storage space
+                    File.Delete(existingReferenceFileBackupPath);
+
+                    createdReference = true;
+                }
+
+                if (!createdReference)
+                {
+                    Log.Trace($"BenzingaDataConverter.WriteToFile(): Skipping news article {article.Id} because no tickers are associated with it");
+                    continue;
+                }
+
+                var finalArticleFile = Path.Combine(finalDirectory.FullName, $"{article.Id}.json");
+                var tempArticleFile = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.json");
+                var backupArticleFile = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.json");
+                var articleExists = File.Exists(finalArticleFile);
+
+                Log.Trace($"BenzingaDataConverter.WriteToFile(): Writing article contents to temp: {tempArticleFile}");
+                File.WriteAllText(tempArticleFile, JsonConvert.SerializeObject(article, Newtonsoft.Json.Formatting.None));
+
+                if (articleExists)
+                {
+                    Log.Trace($"BenzingaDataConverter.WriteToFile(): Article already exists. Moving from: {finalArticleFile} - to backup: {backupArticleFile}");
+                    File.Move(finalArticleFile, backupArticleFile);
+                }
+
+                try
+                {
+                    Log.Trace($"BenzingaDataConverter.WriteToFile(): Moving from temp to: {finalArticleFile}");
+                    File.Move(tempArticleFile, finalArticleFile);
+                }
+                catch (Exception e)
+                {
+                    Log.Error(e, $"BenzingaDataConverter.WriteToFile(): Failed to move file to: {finalArticleFile}");
+
+                    if (articleExists)
+                    {
+                        Log.Error($"BenzingaDataConverter.WriteToFile(): Restoring backup article file: {backupArticleFile} to original location: {finalArticleFile}");
+                        File.Move(backupArticleFile, finalArticleFile);
+                    }
+                }
+            }
+
+            // Now compress all of the data we have for a given day
+            var compressedFinal = new FileInfo(Path.Combine(_destinationDirectory.FullName, "content", $"{date:yyyyMMdd}.zip"));
+            var compressedFinalBackup = new FileInfo(Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.zip"));
```

Comment:
I like seeing you work w/ `FileInfo` -- I used to always just use strings, but keeping things as a `FileInfo` can be very powerful, far superior to holding the value as a string. There's actual an _evil_ in programming called _primitive obsession_ -- it's where you're afraid to create types and instead just pass simple structs everywhere.

Here's a refactoring guru article on [primitive obsession](https://refactoring.guru/smells/primitive-obsession) for anyone following along at home that is interested... well... even if you're not interested the link ain't goin' anywhere 😃